app/api/v2/work/route.js

```diff
@@ -0,0 +1,123 @@
+
+import { NextResponse } from "next/server";
+import prisma from "@/prisma/prismaClient";
+import { Auth } from "../../utils/auth";
+import { getServerSession } from "next-auth";
+import { authOptions } from "@/app/api/utils/authOptions";
+
+
+export async function POST(req) {
+
+  const session = await getServerSession(authOptions)
+  const params = await req.json()
+  const authCheck = new Auth(session, params)
+    .requireRoles([])
+    .requireParams(["loggedBy", "loggedFor", "workedAt", "duration", "description", "semesterId"])
+
+  if (authCheck.failed) return authCheck.verify(authCheck.response)
+
+
+
+  const success = await prisma.$transaction(async (transaction) => {
+    const workLogEntry = await transaction.workLog.create({
+      data: {
+        loggedBy: params.loggedBy,
+        loggedFor: params.loggedFor,
+        workedAt: params.workedAt,
+        duration: params.duration,
+        description: params.description,
+        semesterId: params.semesterId
+      }
+    });
+
+    const voucherBuffer = (await transaction.VoucherBuffer.findFirst({
+      select: {
+        buffer: true
+      },
+      where: {
+        userId: params.loggedFor
+      }
+    })).buffer;
```

Comment:
Det blir error hvis den ikke finner en buffer
```suggestion
    }))?.buffer;
```